pwiz_tools/Skyline/Menus/ChromatogramContextMenu.cs

```diff
@@ -175,6 +178,61 @@ public void BuildChromatogramMenu(ZedGraphControl zedGraphControl, PaneKey paneK
             CopyEmfToolStripMenuItem.AddToContextMenu(zedGraphControl, menuStrip);
         }
 
+        /// <summary>
+        /// Insert ion mobility-related menu items as appropriate
+        /// </summary>
+        private int InsertIonMobilityMenuItems(ToolStripItemCollection items, ChromFileInfoId chromFileInfoId, int iInsert)
+        {
+            var chromFileInfo = DocumentUI.Settings.MeasuredResults?.Chromatograms
+                .Select(chromatogramSet => chromatogramSet.GetFileInfo(chromFileInfoId))
+                .FirstOrDefault(fileInfo => null != fileInfo);
+            if (null != chromFileInfo && chromFileInfo.IonMobilityUnits != eIonMobilityUnits.none)
+            {
+                var asSubMenu = true;
+
+                var ccsMenuItemText = Resources.ChromatogramContextMenu_Collision_Cross_Section;
+                var ccsItem = new ToolStripMenuItem(ccsMenuItemText);
+                ccsItem.Click += (sender, eventArgs) => SkylineWindow.ShowCollisionCrossSection = !SkylineWindow.ShowCollisionCrossSection;
+                ccsItem.Checked = SkylineWindow.ShowCollisionCrossSection;
+
+                string imMenuItemText;
+                switch (chromFileInfo.IonMobilityUnits)
+                {
+                    case eIonMobilityUnits.drift_time_msec:
+                        imMenuItemText = Resources.ChromatogramContextMenu_InsertIonMobilityMenuItems_Drift_Time;
+                        break;
+                    case eIonMobilityUnits.inverse_K0_Vsec_per_cm2:
+                        imMenuItemText = Resources.ChromatogramContextMenu_InsertIonMobilityMenuItems_Inverse_Ion_Mobility;
+                        break;
+                    case eIonMobilityUnits.compensation_V:
```

Comment:
Do we know of any instruments which produce spectra with a true compensation voltage dimension? It feels like our code is starting to become confused about how instruments present compensation voltage.

Reply:
Nevermind, I can see how this might be desirable in the case of extraction from MS1 spectra from Thermo FAIMS DDA.